### PR TITLE
Ctera branch

### PR DESCRIPTION
**Pull request by roberteliass**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by roberteliass** [comment]
@microsoft-github-policy-service agree [company="{CTERA}"]

**Comment by roberteliass** [comment]
@microsoft-github-policy-service agree company="CTERA"

File: Solutions/CTERA/Data Connectors/Logo/CTERA_Networks_logo.svg
Comment (v-visodadasi) [inline Solutions/CTERA/Data Connectors/Logo/CTERA_Networks_logo.svg:1]: Kql Validation Checks are failing for this logo, Can you please check and resolve those validations.
Please correct the id it should be in guid format.

File: Solutions/CTERA/Workbooks/CTERA_Workbook.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      },
+      "name": "query - 2"
+    }
+  ],
+  "fallbackResourceIds": [
+    "/subscriptions/0dfe0fcb-a5a0-4b3b-898d-31c327799113/resourcegroups/product-robert/providers/microsoft.operationalinsights/workspaces/robertsentinel2"
+  ],
+  "fromTemplateId": "sentinel-UserWorkbook",
+  "$schema": "https://github.com/Microsoft/Application-Insights-Workbooks/blob/master/schema/workbook.json"
+}
```
Comment (v-visodadasi) [inline Solutions/CTERA/Workbooks/CTERA_Workbook.json:83]: Please change fromTemplateId, The current value is by default. Please change it.
Also please add workbook metadata to this file https://github.com/Azure/Azure-Sentinel/blob/master/Workbooks/WorkbooksMetadata.json.
Also Share some workbook images with Dark and White background.

**Review by v-visodadasi — CHANGES_REQUESTED** [review]

**Comment by v-visodadasi** [comment]
Please try to resolve the validation errors from your end.
Also share sample data for testing the content of the solution.

**Comment by roberteliass** [comment]
Hi, responding to each section.

**Comment by roberteliass** [comment]
Hi,

I haven’t received any response from you.
Please let me know if there’s any feedback and what are the next steps.

Thanks,
Robert Elias
Product Manager
e. ***@***.******@***.***>
linkedin eliasrobert<http://www.linkedin.com/in/eliasrobert>

***@***.***




From: Robert Elias ***@***.***>
Sent: August 1, 2024 2:06 AM
To: Azure/Azure-Sentinel ***@***.***>; Azure/Azure-Sentinel ***@***.***>
Cc: Mention ***@***.***>
Subject: RE: [Azure/Azure-Sentinel] Ctera branch (PR #10889)

Hi, responding to each section.

**Comment by v-visodadasi** [comment]
Please note that the Solution Metadata file should not be formatted in this manner. For reference, Please see https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/AIShield%20AI%20Security%20Monitoring/SolutionMetadata.json

**Comment by v-visodadasi** [comment]
Please resolve the validation errors

File: Solutions/CTERA/Data/Solution_CTERA.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "Name": "CTERA",
+  "Author": "CTERA Networks - support@ctera.com",
+  "Logo": "<img src=\"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Solutions/CTERA/Data%20Connectors/Logo/CTERA_Networks_logo.svg\" width=\"75px\" height=\"75px\">",
```
[the remark this answers is not in the document]
Comment (roberteliass) [reply Solutions/CTERA/Data/Solution_CTERA.json]: Done

File: Solutions/CTERA/Data Connectors/CTERA_Data_Connector.json [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+              "lastDataReceivedQuery": "Syslog | where Source == 'CTERA' | summarize Time = max(TimeGenerated)"
+            }
+          ],
+          "connectivityCriteria": [
+            {
+              "type": "IsConnectedQuery",
+              "value": "Syslog | where Source == 'CTERA' | summarize LastLogReceived = max(TimeGenerated) | project IsConnected = LastLogReceived > ago(15m)"
+            }
+          ],
+          "availability": {
```
[the remark this answers is not in the document]
Comment (roberteliass) [reply Solutions/CTERA/Data Connectors/CTERA_Data_Connector.json:48]: Done

File: Solutions/CTERA/Workbooks/CTERA_Workbook.json (hunk 1 above)
Comment (roberteliass) [reply Solutions/CTERA/Workbooks/CTERA_Workbook.json:86]: Updated

File: Solutions/CTERA/Data Connectors/Logo/CTERA_Networks_logo.svg
Comment (roberteliass) [reply Solutions/CTERA/Data Connectors/Logo/CTERA_Networks_logo.svg:1]: Resolved resolution issue

**Comment by roberteliass** [comment]
> Please try to resolve the validation errors from your end. Also share sample data for testing the content of the solution.

I can't understand why the validation is failing neither on the logo, nor for the rest of the validations.
Regarding the sample data, can I supply the syslog events?

**Comment by roberteliass** [comment]
I’m not sure I understand why the validation fails, is there a clear way to identify it?

Thanks,
Robert Elias
Product Manager
e. ***@***.******@***.***>
linkedin eliasrobert<http://www.linkedin.com/in/eliasrobert>

***@***.***

**Comment by v-prasadboke** [comment]
Hello @roberteliass, Can you provide me write access to your branch. 
Need to push some required commits. 

You can do this by inviting me as an contributor to your branch. 

Thanks, 
Prasad

**Comment by roberteliass** [comment]
> Hello @roberteliass, Can you provide me write access to your branch. Need to push some required commits.
> 
> You can do this by inviting me as an contributor to your branch.
> 
> Thanks, Prasad

I provided the access, let me know if you have it.

In addition please let me know what is the status on my request to better understand the validation failures.
When I try seeing the log, I can't have access to it.

File: Solutions/CTERA/Data Connectors/CTERA_Data_Connector.json [hunk 4]
```diff
@@ -0,0 +1,133 @@
+{
+    "id": "CTERA",
+    "title": "CTERA Syslog",
+    "publisher": "CTERA",
+    "description": "Ingest CTERA syslog data into Microsoft Sentinel",
+    "additionalRequirementBanner": "None ",
```
Comment (v-prasadboke) [inline Solutions/CTERA/Data Connectors/CTERA_Data_Connector.json:6]: Hello @roberteliass, Please add a short description markdown to the data connector. 
For ref: https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Theom/Data%20Connectors/Theom.json

**Comment by v-prasadboke** [comment]
Please do share sample data to test the content of the solution

File: Solutions/CTERA/Data Connectors/CTERA_Data_Connector.json (hunk 4 above)
Comment (roberteliass) [reply Solutions/CTERA/Data Connectors/CTERA_Data_Connector.json:6]: Updated, will commit shortly

**Comment by roberteliass** [comment]
Please find the syslog messages files which were used for testing the solution in this link:
https://cti.ctera.com/invitations/?share=c96e27fca1369ea917d8&dl=1

Thanks,
Robert Elias
Product Manager
e. ***@***.******@***.***>
linkedin eliasrobert<http://www.linkedin.com/in/eliasrobert>

***@***.***

**Comment by roberteliass** [comment]
@v-prasadboke I performed a rebase from master to the CTERA branch, and although the changes you committed are there but still the WorkbooksValidations and the DetectionTemplateSchemaValidation are failing for some reason.
Can you assist?

**Comment by roberteliass** [comment]
@v-prasadboke and @v-visodadasi I would appreciate your help to overcome the failing validations.

**Pull request closed without merging**